API/Services/BookService.cs

```diff
@@ -425,10 +426,189 @@ private async Task InlineStyles(HtmlDocument doc, EpubBookRef book, string apiBa
+    {
+        return (doc.DocumentElement?.SelectSingleNode(path + "//rdf:li", ns)
+            ?? doc.DocumentElement?.SelectSingleNode(path, ns))?.InnerText;
+    }
+
+    private float? GetFloatFromXmlNode(XmlDocument doc, XmlNamespaceManager ns, string path)
+    {
+        var text = GetTextFromXmlNode(doc, ns, path);
+        if (string.IsNullOrEmpty(text)) return null;
+
+        return float.Parse(text);
+    }
+
+    private string? GetListFromXmlNode(XmlDocument doc, XmlNamespaceManager ns, string path)
+    {
+        var nodes = doc.DocumentElement?.SelectNodes(path+"//rdf:li", ns);
+        if (nodes == null) return null;
+        var list = new StringBuilder();
+        foreach (XmlNode n in nodes)
+        {
+            if (list.Length > 0)
+            {
+                list.Append(",");
+            }
+            list.Append(n.InnerText);
+        }
+        return list.Length > 0 ? list.ToString() : null;
+    }
+
+    private DateTime? GetDateTimeFromXmlNode(XmlDocument doc, XmlNamespaceManager ns, string path)
+    {
+        var text = GetTextFromXmlNode(doc, ns, path);
+        if (text == null) return null;
+
+        return DateTime.Parse(text);
+    }
+
+    private ComicInfo? GetPdfComicInfoFromMetadata(string metadata, string filePath)
+    {
+        var metaDoc = new XmlDocument();
+        metaDoc.LoadXml(metadata);
+
+        var ns = new XmlNamespaceManager(metaDoc.NameTable);
+        ns.AddNamespace("rdf", "http://www.w3.org/1999/02/22-rdf-syntax-ns#");
+        ns.AddNamespace("dc", "http://purl.org/dc/elements/1.1/");
+        ns.AddNamespace("calibreSI", "http://calibre-ebook.com/xmp-namespace-series-index");
+        ns.AddNamespace("calibre", "http://calibre-ebook.com/xmp-namespace");
+        ns.AddNamespace("pdfx", "http://ns.adobe.com/pdfx/1.3/");
+        ns.AddNamespace("prism", "http://prismstandard.org/namespaces/basic/2.0/");
+        ns.AddNamespace("xmp", "http://ns.adobe.com/xap/1.0/");
+
+        var info = new ComicInfo();
+        var publicationDate = GetDateTimeFromXmlNode(metaDoc, ns, "//dc:date")
+            ?? GetDateTimeFromXmlNode(metaDoc, ns, "//xmp:createdate");
+        if (publicationDate != null)
+        {
+            info.Year  = publicationDate.Value.Year;
+            info.Month = publicationDate.Value.Month;
+            info.Day   = publicationDate.Value.Day;
+        }
+        info.Summary   = GetTextFromXmlNode(metaDoc, ns, "//dc:description") ?? "";
+        info.Publisher = GetTextFromXmlNode(metaDoc, ns, "//dc:publisher") ?? "";
+        info.Writer    = GetListFromXmlNode(metaDoc, ns, "//dc:creator") ?? "";
+        info.Title     = GetTextFromXmlNode(metaDoc, ns, "//dc:title") ?? "";
+        info.Genre     = GetListFromXmlNode(metaDoc, ns, "//dc:subject") ?? "";
+        info.LanguageISO = ValidateLanguage(GetTextFromXmlNode(metaDoc, ns, "//dc:language"));
+        ComicInfo.CleanComicInfo(info);
```

Comment:
Clean should be last

Reply:
I was following the structure of the corresponding epub implementation. Should clean in that one also be moved?